netmiko/ipinfusion/ipinfusion_ocnos.py

```diff
@@ -28,15 +28,156 @@ def session_preparation(self) -> None:
         self._test_channel_read()
         self.set_base_prompt()
         self.disable_paging(command="terminal length 0")
-
+        # Turn off logging for the session as it can spoil analyzing of outputs
+        self.send_command("terminal no monitor")
         # Clear the read buffer
         time.sleep(0.3 * self.global_delay_factor)
         self.clear_buffer()
 
+    def send_config_set(self, *args: Any, **kwargs: Any) -> str:
+        """Send config command(s). Requires separate calling of commit to apply."""
+
+        # Do not exit config mode - it won't allow until you commit or abort transaction
+        output = super().send_config_set(*args, **kwargs, exit_config_mode=False)
+        error_markers = [
+            "% Invalid input detected at '^' marker.",
+            "% Parameter not configured",
+            "% Ambiguous command:",
+        ]
+        for error_marker in error_markers:
+            if error_marker in output:
+                raise ValueError(
+                    f"Send config set failed with the following errors:\n\n{output}"
+                )
+        return output
+
+    def commit(
+        self,
+        confirm: bool = False,
+        confirm_delay: Optional[int] = None,
+        description: str = "",
+        read_timeout: float = 120.0,
+    ) -> str:
+        """
+        Commit the candidate configuration.
+
+        default (no options):
+            command_string = commit
+        confirm and confirm_delay:
+            command_string = commit confirmed <confirm_delay>
```

Comment:
We should remove `commit confirm` from the `commit` method. These should be implemented as a private method as per the communication in https://github.com/ktbyers/netmiko/issues/3590

Reply:
Same with `confirm_delay` this argument should be moved to a private `_commit_confirm()` method.

Reply:
Or worded slightly differently, Netmiko's `commit()` method only commits previous changes--it does not support `commit confirm`, this would need to be done entirely in private methods.